### PR TITLE
Ensure proper non-dev version string when publishing to PyPI

### LINKED ISSUE #901: Issues of the "Publish to PyPI" workflow
**Description of the problem**

The [`publish-to-pypi.yml`](https://github.com/GenericMappingTools/pygmt/blob/master/.github/workflows/publish-to-pypi.yml) workflow uploads the PyGMT packages to PyPI and/or TestPyPI.

During the release v0.3.0, we found an issue with the workflow and had a quick fix in #900. 

When I reiviwed the workflow, I found two more issues:

1. We're using secrets `test_pypi_password` and `pypi_password` in the `pypa/gh-action-pypi-publish` action. 
https://github.com/GenericMappingTools/pygmt/blob/65b099a5873b4f9e76d7fccdbc8882d9b37870e3/.github/workflows/publish-to-pypi.yml#L54-L64

The names of these two secrets are confusing. They are API tokens (generated by me, if I remember correctly), not someone's passwords. We should rename them to better names, `TEST_PYPI_API_TOKEN` and `PYPI_API_TOKEN` following the [official guide](https://packaging.python.org/guides/publishing-package-distribution-releases-using-github-actions-ci-cd-workflows/).

2. The workflow always fails on forks, because forks don't have these secrets. We should skip the workflow for forks.

### PR DESCRIPTION
**Pull request by weiji14**
**Description of proposed changes**

Ensures that the PyPI release Github Action workflow has a proper version string like `v0.3.0` instead of `v0.3.1.dev0`. Also edit the Publish-to-PyPI workflow to only run on GenericMappingTools/pygmt and rename secrets from `*_PYPI_PASSWORD` to `*_PYPI_API_TOKEN`

Publish Publish Github Action Documentation is at https://github.com/pypa/gh-action-pypi-publish/tree/v1.4.2

Addresses https://github.com/GenericMappingTools/pygmt/issues/845#issuecomment-779148771, also fixes #901.

**Reminders**

- [ ] Run `make format` and `make check` to make sure the code follows the style guide.
- [ ] Add tests for new features or tests that would have caught the bug that you're fixing.
- [ ] Add new public functions/methods/classes to `doc/api/index.rst`.
- [ ] Write detailed docstrings for all functions/methods.
- [ ] If adding new functionality, add an example to docstrings or tutorials.

**Slash Commands**

You can write slash commands (`/command`) in the first line of a comment to perform
specific operations. Supported slash commands are:

- `/format`: automatically format and lint the code
- `/test-gmt-dev`: run full tests on the latest GMT development version

### CODE REVIEW

File: .github/workflows/publish-to-pypi.yml [hunk 1]
```diff
@@ -36,6 +36,7 @@ jobs:
       run: python -m pip install setuptools wheel
 
     - name: Fix up version string
+      if: !startsWith(github.ref, 'refs/tags')
```
Comment (seisman) [inline .github/workflows/publish-to-pypi.yml:39]: Can you also move the comment "The step is only necessary for testing purpose" here, so we can better why we use `!startsWith(github.ref, 'refs/tags')`.

**Comment by seisman** [comment]
Please see #901, and let me know if you want to fix it in this PR.

File: .github/workflows/publish-to-pypi.yml (hunk 1 above)
Comment (weiji14) [reply .github/workflows/publish-to-pypi.yml]: Ok, done at 82290d66716a097d12c135ad713f2f8f3f8057c9

**Comment by weiji14** [comment]
> Please see #901, and let me know if you want to fix it in this PR.

Ok, I think the issues in #901 have been addressed, but please double check. **Note**: Will need to rename `TEST_PYPI_PASSWORD` -> `TEST_PYPI_API_TOKEN` and `PYPI_PASSWORD` -> `PYPI_API_TOKEN` before merging this PR.

File: .github/workflows/publish-to-pypi.yml [hunk 2]
```diff
@@ -35,13 +36,13 @@ jobs:
     - name: Install dependencies
       run: python -m pip install setuptools wheel
 
-    - name: Fix up version string
+    # This step is only necessary for testing purposes and for TestPyPI
+    - name: Fix up version string for TestPyPI
+      if: ${{ !startsWith(github.ref, 'refs/tags') }}
```
Comment (seisman) [inline .github/workflows/publish-to-pypi.yml:41]: If I understand it correctly, when we make a release/tag, this sed command won't run. So, we are pushing the same versions (e.g., `v0.3.1`) to both PyPI and TestPyPI, right?

Comment (weiji14) [reply .github/workflows/publish-to-pypi.yml:41]: Yes, correct. But I think pushing v0.3.1 to both PyPI and TestPyPI is fine right?

Comment (seisman) [reply .github/workflows/publish-to-pypi.yml:41]: Yes, it looks good.

**Review by seisman — APPROVED** [review]
Things to do before merging:

1. Rename the secrets in https://github.com/GenericMappingTools/pygmt/settings/secrets/actions
2. Update the PR title

**Comment by weiji14** [comment]
> Things to do before merging:
> 
>     1. Rename the secrets in https://github.com/GenericMappingTools/pygmt/settings/secrets/actions

Ok, generated new tokens and created new secrets for PyPI and TestPyPI. Also deleted the old `TEST_PYPI_PASSWORD` and `PYPI_PASSWORD` secrets at https://github.com/GenericMappingTools/pygmt/settings/secrets/actions.

>     2. Update the PR title

See if it looks ok, having trouble describing this PR :sweat_smile:

**Review by seisman — APPROVED** [review]
Looks good.

**Comment by weiji14** [comment]
I might trigger a quick test (to TestPyPI) just to make sure things work.

**Pull request merged**